astropy/io/fits/fitstime.py

```diff
@@ -362,6 +362,10 @@ def _convert_time_column(col, column_info):
             # [+/-C]CCYY-MM-DD[Thh:mm:ss[.s...]] where the number of characters
             # from index 20 to the end of string represents the precision
             precision = max(int(col.info.dtype.str[2:]) - 20, 0)
+            if col.info.dtype.kind == 'S':
```

Comment:
I'd not be against solving this on the `Time` side - @taldcroft - what do you think? It would not be difficult (but definitely in another PR).

Reply:
Actually, `col` here is an astropy `Column`, which should sandwich the bytes already, so I think that this doesn't work automatically is a buglet.

Reply:
See #6823 for a fix. I think it is fairly trivial, so perhaps we can merge it before this PR, so that this change no longer is necessary.

Reply:
@mhvk - sounds good!

Reply:
#6823 is merged now.

Reply:
@taldcroft @mhvk - interestingly I still need these lines otherwise the tests fail, so there must still be some subtleties. Maybe one of you could try and run the tests for fitstime on this branch locally and see what might be missing for Time? In the mean time, I'll leave as-is.

Reply:
It is a bit strange - it has to do that the sample columns put in that file have different length:
```
col
# <Column name='datetime' dtype='bytes29' length=2>
# 1999-01-01T00:00:00.123456789
#           2010-01-01T00:00:00
Time(col[0], format='fits', precision=9)
# <Time object: scale='utc' format='fits' value=1999-01-01T00:00:00.123456789(UTC)>
Time(col[1], format='fits', precision=9)
# <Time object: scale='utc' format='fits' value=2010-01-01T00:00:00.000000000(UTC)>
Time(col, format='fits', precision=9)
# *** ValueError: Input values did not match the format class fits
```

Reply:
the example is not good (`col` translates to string for items), but the issue is there...

Reply:
See #6903

Reply:
Don't forget to remove this stanza!